src/main/java/com/taskadapter/redmineapi/internal/Transport.java

```diff
@@ -254,7 +255,10 @@ public User getCurrentUser(NameValuePair... params) throws RedmineException {
 		final URI uri = getURIConfigurator().getObjectURI(obj.getClass(),
 				Integer.toString(obj.getId()));
 		final HttpPut http = new HttpPut(uri);
-
+                if (getImpersonateUser() != null) {
+                   http.addHeader("X-Redmine-Switch-User", getImpersonateUser());
+                }
+  
```

Comment:
so why is this new user used only in "update object" method? what about all others?